### PR TITLE
Add support for routing_mode to google_network_services_gateway

### PR DESCRIPTION
**Pull request by Samir-Cit**
Hello folks.

This PR is to add the support for Secure Web Proxy as a Next Hop by adding `routing_mode` field with possible values `EXPLICIT_ROUTING_MODE` and `NEXT_HOP_ROUTING_MODE` to the `google_network_services_gateway` resource

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
networkservices: added `routing_mode` field to `google_network_services_gateway` resource
```

### CODE REVIEW

**Comment by zli82016** [comment]
Hello, @Samir-Cit , adding a required field is a breaking change in Terraform provider and is not allowed in the provider minor version release. But the new field can be added in the next major release (7.0) of providers.

This is the [doc](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes/). Feel free to let me know if you have any questions.

Also, is there Github issue for this PR? Thanks.

**Comment by Samir-Cit** [comment]
> Hello, @Samir-Cit , adding a required field is a breaking change in Terraform provider and is not allowed in the provider minor version release. But the new field can be added in the next major release (7.0) of providers.
> 
> This is the [doc](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes/). Feel free to let me know if you have any questions.
> 
> Also, is there Github issue for this PR? Thanks.

Hello @zli82016 
Got it, there's no problem this change go in the next major release.

I don't believe we have a Github issue for this. I work with @Naveen and this task was on our backlog.
But I'll double check and ask him, if we have I'll put it here!

**Comment by zli82016** [comment]
> > Hello, @Samir-Cit , adding a required field is a breaking change in Terraform provider and is not allowed in the provider minor version release. But the new field can be added in the next major release (7.0) of providers.
> > This is the [doc](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes/). Feel free to let me know if you have any questions.
> > Also, is there Github issue for this PR? Thanks.
> 
> Hello @zli82016 Got it, there's no problem this change go in the next major release.
> 
> I don't believe we have a Github issue for this. I work with @naveen and this task was on our backlog. But I'll double check and ask him, if we have I'll put it here!

Thank you for your understanding.

If there isn't a Github issue, feel free to create one, which will help to track it in the next major release. 

Feel free to close this PR. Thanks.

File: mmv1/products/networkservices/Gateway.yaml [hunk 1]
```diff
@@ -206,3 +207,12 @@ properties:
       This feature only applies to gateways of type 'SECURE_WEB_GATEWAY'.
     item_type:
       type: String
+  - name: 'routingMode'
+    type: Enum
+    immutable: true
+    description: |
+      The routing mode of the Gateway. This field is configurable only for gateways of type SECURE_WEB_GATEWAY. This field is required for gateways of type SECURE_WEB_GATEWAY.
+    custom_flatten: 'templates/terraform/custom_flatten/network_services_gateway_routing_mode.go.tmpl'
```
Comment (zli82016) [inline mmv1/products/networkservices/Gateway.yaml:215]: Is the `custom_flatten` needed here? It causes the terraform plan not empty.

File: mmv1/templates/terraform/pre_create/network_services_gateway.go.tmpl [hunk 2]
```diff
@@ -0,0 +1,3 @@
+if obj["type"] == "SECURE_WEB_GATEWAY" && obj["routingMode"] == nil {
+    return fmt.Errorf("Field routing_mode is required for Gateway of type SECURE_WEB_GATEWAY")
```
Comment (zli82016) [inline mmv1/templates/terraform/pre_create/network_services_gateway.go.tmpl:2]: 1. Does the server side have the similar validation when creating a new resource? If yes, the validation is not needed in the Terraform provider.

2. Does the server side have the similar validation when updating other fields in the resource?

File: mmv1/products/networkservices/Gateway.yaml (hunk 1 above)
Comment (Samir-Cit) [reply mmv1/products/networkservices/Gateway.yaml:215]: It's not

File: mmv1/templates/terraform/pre_create/network_services_gateway.go.tmpl (hunk 2 above)
Comment (Samir-Cit) [reply mmv1/templates/terraform/pre_create/network_services_gateway.go.tmpl:2]: Not needed

File: mmv1/third_party/terraform/services/networkservices/resource_network_services_gateway_test.go [hunk 3]
```diff
@@ -358,6 +360,7 @@ resource "google_network_services_gateway" "foobar" {
   location                             = "us-east1"
   addresses                            = ["10.128.0.99"]
   type                                 = "SECURE_WEB_GATEWAY"
+  routing_mode                         = "EXPLICIT_ROUTING_MODE"
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/networkservices/resource_network_services_gateway_test.go:363]: Can you please remove this line to test if the default value handing is working with `default_from_api: true` ?

Comment (zli82016) [reply mmv1/third_party/terraform/services/networkservices/resource_network_services_gateway_test.go:363]: Can you please remove this line to see if the test will pass without the new field?

We don't want to add the optional field to all of the test cases.

Comment (Samir-Cit) [reply mmv1/third_party/terraform/services/networkservices/resource_network_services_gateway_test.go:363]: Done

File: mmv1/templates/terraform/examples/network_services_gateway_secure_web_proxy.tf.tmpl [hunk 4]
```diff
@@ -51,6 +51,7 @@ resource "google_network_services_gateway" "{{$.PrimaryResourceId}}" {
   location                             = "us-central1"
   addresses                            = ["10.128.0.99"]
   type                                 = "SECURE_WEB_GATEWAY"
+  routing_mode                         = "EXPLICIT_ROUTING_MODE"
```
Comment (zli82016) [inline mmv1/templates/terraform/examples/network_services_gateway_secure_web_proxy.tf.tmpl:54]: Do you mind removing all of the `EXPLICIT_ROUTING_MODE` in the tests?

Comment (Samir-Cit) [reply mmv1/templates/terraform/examples/network_services_gateway_secure_web_proxy.tf.tmpl]: Done

**Review by zli82016 — COMMENTED** [review]
LGTM. Thanks.

**Review by zli82016 — APPROVED** [review]

**Pull request merged**